homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,268 @@
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    PRESET_AWAY,
+    PRESET_ECO,
+    PRESET_HOME,
+    PRESET_NONE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER_NAME
+from .coordinator import CompitConfigEntry, CompitDataUpdateCoordinator
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+# Device class for climate devices in Compit system
+CLIMATE_DEVICE_CLASS = 10
+
+# Coordinator is used to centralize the data updates
+PARALLEL_UPDATES = 0
+
+COMPIT_MODE_MAP = {
+    CompitHVACMode.COOL: HVACMode.COOL,
+    CompitHVACMode.HEAT: HVACMode.HEAT,
+    CompitHVACMode.OFF: HVACMode.OFF,
+}
+
+COMPIT_FANSPEED_MAP = {
+    CompitFanMode.OFF: FAN_OFF,
+    CompitFanMode.AUTO: FAN_AUTO,
+    CompitFanMode.LOW: FAN_LOW,
+    CompitFanMode.MEDIUM: FAN_MEDIUM,
+    CompitFanMode.HIGH: FAN_HIGH,
+    CompitFanMode.HOLIDAY: FAN_AUTO,
+}
+
+COMPIT_PRESET_MAP = {
+    CompitPresetMode.AUTO: PRESET_HOME,
+    CompitPresetMode.HOLIDAY: PRESET_ECO,
+    CompitPresetMode.MANUAL: PRESET_NONE,
+    CompitPresetMode.AWAY: PRESET_AWAY,
+}
+
+HVAC_MODE_TO_COMPIT_MODE = {v: k for k, v in COMPIT_MODE_MAP.items()}
+FAN_MODE_TO_COMPIT_FAN_MODE = {v: k for k, v in COMPIT_FANSPEED_MAP.items()}
+PRESET_MODE_TO_COMPIT_PRESET_MODE = {v: k for k, v in COMPIT_PRESET_MAP.items()}
+
+ATTR_TEMPERATURE = "temperature"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: CompitConfigEntry,
+    async_add_devices: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the CompitClimate platform from a config entry."""
+
+    coordinator: CompitDataUpdateCoordinator = entry.runtime_data
+
+    climate_entities = []
+    for device_id in coordinator.connector.devices:
+        device = coordinator.connector.devices[device_id]
+
+        if device.definition.device_class == CLIMATE_DEVICE_CLASS:
+            climate_entities.append(
+                CompitClimate(
+                    coordinator,
+                    device_id,
+                    {
+                        parameter.parameter_code: parameter
+                        for parameter in device.definition.parameters
+                    },
+                    device.definition.name,
+                )
+            )
+
+    async_add_devices(climate_entities)
+
+
+class CompitClimate(CoordinatorEntity[CompitDataUpdateCoordinator], ClimateEntity):
+    """Representation of a Compit climate device."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.COOL, HVACMode.OFF]
```

Comment:
```suggestion
    _attr_hvac_modes = [*COMPIT_MODE_MAP.values()]
```
that way they are consistent